src/demo/player.rs

```diff
@@ -88,11 +86,24 @@ fn add_player_movement_on_spawn(
 fn record_player_directional_input(
     trigger: Trigger<Fired<PlayerMove>>,
     movement_controller: Single<(&mut TnuaController, &MovementSettings)>,
+    camera_query: Single<&Transform, With<Camera3d>>,
+    time: Res<Time>,
 ) {
     let (mut controller, settings) = movement_controller.into_inner();
+    let camera_transform = camera_query.into_inner();
+    let mut camera_right = camera_transform.right().as_vec3();
+    let mut camera_forward = camera_transform.forward().as_vec3();
+    camera_right.y = 0.0;
+    camera_forward.y = 0.0;
```

Comment:
very small nit but lets you keep them immutable at the cost of 1 extra vec3

```suggestion
    let camera_right = camera_transform.right().as_vec3().with_y(0.0);
    let camera_forward = camera_transform.forward().as_vec3().with_y(0.0);
```

Reply:
good to know!